core/commands/inline_diff.py

```diff
@@ -219,26 +219,33 @@ def get_object_contents(self, object_hash):
         """
         return self.git("show", "--no-color", object_hash)
 
+    def get_file_contents_binary(self, file_path):
+        """
+        Given an absolute file path, return the binary contents of that file
+        as a string.
+        """
+        file_path = os.path.join(self.repo_path, file_path)
+        with open(file_path, "rb") as f:
+            return f.read()
+
     def get_file_contents(self, file_path):
         """
         Given an absolute file path, return the text contents of that file
         as a string.
         """
-        file_path = os.path.join(self.repo_path, file_path)
+        binary = self.get_file_contents_binary(file_path)
         try:
-            with open(file_path, "rt", encoding="utf-8") as f:
-                return f.read()
+            return binary.decode('utf-8')
         except UnicodeDecodeError as unicode_err:
-            with open(file_path, "rt", encoding="latin-1") as f:
-                return f.read()
+            return binary.decode('latin-1')
 
     def get_object_from_string(self, string):
         """
         Given a string, pipe the contents of that string to git and have it
         stored in the current repo, and return an object-hash that can be
         used to diff against.
         """
-        stdout = self.git("hash-object", "-w", "--stdin", stdin=string)
+        stdout = self.git("hash-object", "-w", "--stdin", stdin=string, encode=False)
```

Comment:
These should be cleaned up by `gc` whenever their creation date falls before the `prune` date.  You can force that by passing an explicit date or `all`.

As to why it is here, this is used in a couple of contexts for content that is not currently committed to the repo anywhere.  I can't recall why it was needed here, but even in large repos, I have not noticed any performance impact from the orphaned git blobs.